### DISCUSSION TITLE
Landing with velocity mode

### DISCUSSION THREAD

**Discussion by puliceworx** [bitcraze/discussions#915] [Q&A] [answered]
Let me begin by stating up front that I have debugged and figured out how CFClient lands in assisted mode and I assume if we did it the same way we would not have the issue we are having.  With that stated you can choose to skip this or not.  :) 

If you're still reading this is the problem: 
LAND setpoint is velocity and z position = 0 (this is where it differs A LOT from cfclient)
Enter Flying -> Warning, level out
Enter Warning, level out -> Exception, free fall
Enter Exception, free fall -> Locked
Locked, reboot required

I've been staring at the Supervisor code to figure out a way around or out of this without changing the 
flight control loop because I did not write that portion.  Nonetheless, I've got to figure it out.  

My suggestion was to change the landing method to something similar to what cfclient does.
I assume crazyflie lands without all of these supervisor state warnings because its let down easy 
until thrust is at a point to back to 0 thrust  at which time stop setpoints are sent.

**Comment by krichardsson** [comment, marked as answer]
The problem is probably that you do not continuously send setpoints to the Crazyflie. There is a timeout mechanism in the Crazyflie that kicks in when communication is lost, that is if no set points are received. Simply resend the same setpoint 3-4 times per second. 

You might also be interested in https://github.com/bitcraze/crazyflie-firmware/issues/1297 to be able to take off again.

**Reply by puliceworx** [reply]
Ah ha, that other thread made a lot of sense with regard to hl commander setpoints being shut off for priority.  

Thanks much, I will try both to see which is a good solution for us.  We send every 30ms while flying but on landing, I know they stop sending setpoints.

**Comment by puliceworx** [comment]
For anyone interested...

Either method works.  For us, the easier method was to just relax the priority as stated in the discussion posted in the answer.  We are not running from a script so we call 
`commanderRelaxPriority();
`

Problem solved, thank you!!
